### ISSUE TITLE
Stash time in the fluid state?

### ISSUE THREAD

**Issue by MTCam** [illinois-ceesd/mirgecom#664]
This is a feature ~~request~~ suggestion, mostly to see if it is laughed out of the room.
 
Can we stash time somewhere in the fluid state?  That'd be great? 🤔

_Originally posted by @MTCam in https://github.com/illinois-ceesd/mirgecom/pull/648#discussion_r868597455_

**Comment by majosm** [comment]
One alternative would be to define the BCs inside the RHS. Then you wouldn't need to pass the `time` through the operators, you could just evaluate the boundary solutions at the current time and set up the BCs accordingly before passing them in.

**Comment by inducer** [comment]
@majosm beat me to suggesting that. :)

**Comment by MTCam** [comment]
I think I'm unclear on the details of how that would work.  Wouldn't that get a a re-compile at every substep?

**Comment by inducer** [comment]
No, I don't think so. Time comes into the RHS as a placeholder for an argument. You `partial` that into the relevant BC functions and pass those into the operators. Nothing should trigger a recompile there IMO.
